glib-macros/src/lib.rs

```diff
@@ -153,36 +173,45 @@ use utils::{parse_nested_meta_items_from_stream, NestedMetaItem};
 ///
 /// You can do it in two different ways:
 ///
-/// Either by providing the value yourself using `@default-return`:
+/// Either by providing the value yourself using `#[default_return]`:
 ///
 /// ```
 /// use glib;
 /// use glib_macros::clone;
 /// use std::rc::Rc;
 ///
 /// let v = Rc::new(1);
-/// let closure = clone!(@weak v => @default-return false, move |x| {
-///     println!("v: {}, x: {}", v, x);
-///     true
-/// });
+/// let closure = clone!(
+///     #[weak] v,
+///     #[default_return]
+///     || false,
+///     move |x| {
+///         println!("v: {}, x: {}", v, x);
+///         true
+///     },
+/// );
 ///
 /// // Drop value so that the weak reference can't be upgraded.
 /// drop(v);
 ///
 /// assert_eq!(closure(2), false);
 /// ```
 ///
-/// Or by using `@default-panic` (if the value fails to get upgraded, it'll panic):
+/// Or by using `#[default_return(panic)]` (if the value fails to get upgraded, it'll panic):
 ///
 /// ```should_panic
 /// # use glib;
 /// # use glib_macros::clone;
 /// # use std::rc::Rc;
 /// # let v = Rc::new(1);
-/// let closure = clone!(@weak v => @default-panic, move |x| {
-///     println!("v: {}, x: {}", v, x);
-///     true
-/// });
+/// let closure = clone!(
+///     #[weak] v,
+///     #[default_return(panic)]
```

Comment:
Are there opinions on this? Should it be called something like `#[default_on_weak_upgrade_failure(panic)]` and `#[default_on_weak_upgrade_failure(return)]` or so instead?

Reply:
Simpler is better I think so as is looks good to me.

Reply:
Not strong opinion, but I would prefer to have `clone_or!(false, ...` and `clone_or_panic!` variants and do not allow `#[weak]` in the simple `clone!`.

Reply:
Having separate variants seems like something we'll repeatedly have to tell people about when they ask why `#[weak]` does not work

Reply:
how about `#[on_upgrade_failure(return)]` ? `default` doesnt mean anything in this context...

Reply:
I'd prefer something shorter. We'll have some time for bikeshedding here before this is merged :)

Reply:
I prefer the variant you list under `my preference`. Also over what is implemented here right now. Unless there are more opinions, I'll change it to that and implement the two other variants you mentioned.

Reply:
As discussed on Matrix that might be too implicit and cause `false` to be returned unintentionally.

Reply:
`upgrade_or_panic` seems most descriptive to me

Reply:
So let's summarize: `#[upgrade_or_else]` for the closure, `#[upgrade_or]` for a plain value (needs to be `Copy`!), `#[upgrade_or_default]` for returning `Default::default()`, `#[upgrade_or_panic]` for panicking and if nothing is specified then `()` is returned (and it will fail to compile if the closure is supposed to return a different type).